lib/servers/Example_Serial_Server.py

```diff
@@ -0,0 +1,138 @@
+# Created on Feb 12, 2017
+# @author: Anthony Ransford
+
+# The following is information for the node server in order to register
+# this server as available to the node. We generally use the same name
+# for instancename and name
+
+"""
+### BEGIN NODE INFO
+[info]
+name = Example_Serial_Server
+version = 1.0
+description = example serial server
+instancename = Example_Serial_Server
+[startup]
+cmdline = %PYTHON% %FILE%
+timeout = 20
+[shutdown]
+message = 987654321
+timeout = 20
+### END NODE INFO
+"""
+
+from labrad.types import Value
+from labrad.devices import DeviceServer, DeviceWrapper
+from labrad.server import setting
+from twisted.internet.defer import inlineCallbacks, returnValue
+
+TIMEOUT = Value(5, 's')  # serial read timeout
+BAUDRATE = 115200
+
+
+class dummyDevice(DeviceWrapper):
+
+    """
+    This class is called by the server class (below) for each device that is connected
+    this allows a single server to talk to different identical devices
+    """
+
+    @inlineCallbacks
+    def connect(self, server, port):
+        """Here we make a connection to the serial server in LabRAD where all
+        the serial communication is handled"""
+        print 'connecting to "%s" on port "%s"...' % (server.name, port),
+        self.server = server
+        self.ctx = server.context()  # grabs an identification number from the server
+        self.port = port
+        # The following opens a communication on a com port specifying a baudrate and a timeout
+        p = self.packet()
+        p.open(port)
+        p.baudrate(BAUDRATE)
+        p.read()  # clear out the read buffer
+        p.timeout(TIMEOUT)
+        yield p.send()
+
+    def packet(self):
```

Comment:
this is repetition of the stepper code.  Were you intending to inherit this code for the stepper motor server?